### DISCUSSION TITLE
Solution to 13-pin ST7565 to Arduino MEGA2560 using u8g2 sketches

### DISCUSSION THREAD

**Discussion by Gudzys** [olikraus/u8g2#2052] [Q&A] [answered]
What worked for this 13-pin ST7565 backlit LCD:

Pin 1 (chip select) to Arduino D10
Pin 2 (reset) to Arduino D8
Pin 3 (register select) to Arduino D9
Pin 4 (SCL) to  to Arduino D11
Pin 5 (SI) to Arduino D13
Pin 8 (3.3v) to Arduino 3.3v
Pin 9 (GND) to Arduino GND

Uncommented this line: U8G2_ST7565_64128N_F_4W_SW_SPI u8g2(U8G2_R0, /* clock=*/ 13, /* data=*/ 11, /* cs=*/ 10, /* dc=*/ 9, /* reset=*/ 8);

In the prepare section: 
    u8g2.setContrast(220);
    u8g2.setFlipMode(1);

![20221215_160117](https://user-images.githubusercontent.com/119560681/207996023-fb31ecbd-4815-4720-8663-59a6a834f0dd.jpg)

**Comment by olikraus** [comment, marked as answer]
Clock is SCL, data pin  is SI.
I assume the display requires 3.3V, so power and signal pins must be 3.3v
If your Arduino is a 5V version, then you need an HC4050 for the signals.

**Reply by Gudzys** [reply]
The Arduino Mega2560 has a 3.3v output and the LCD ST7565 has a 3.3v input (pin 6 VDD); is it too much to hope that connecting these 2 pins and also connecting pin 7 VSS gnd to Arduino gnd is sufficient to power the LCD ST7565?

Also: pins 8 A & 9 K are supposed to be 3.3v & gnd and/or vis-versa. Do these need to be used?

Thanks again for your help.

**Reply by olikraus** [reply]
Sure, power pins are ok

**Reply by Gudzys** [reply]
The sketch is running on the device but the back-lighting is so strong that the characters & graphics are barely visible. Am running GraphicsTest now. Any clue about how to reduce the back-lighting?

Had to link pin 8 to 3.3v and 9 to gnd for it to power. 

Thanks again for your help. Really appreciated.

**Reply by olikraus** [reply]
Maybe you need to adjust the contrast with the SetContrast command in u8g2.

**Reply by Gudzys** [reply]
Yes, that worked. Thanks again for all of your help. Merry Christmas or whatever you celebrate, OK. 

Brent

**Reply by olikraus** [reply]
Christmas preparation already started for me as well 🤗. Merry Christmas.

**Comment by Gudzys** [comment]
Thank you.


------- Original Message -------

**Comment by Gudzys** [comment]
Thank you.


------- Original Message -------
